news/utils.py

```diff
@@ -118,36 +119,36 @@ def lookup_subscriber(token=None, email=None):
         kwargs['token'] = token
     if email:
         kwargs['email'] = email
-    user_data = None
-    try:
-        subscriber = Subscriber.objects.get(**kwargs)
-    except Subscriber.DoesNotExist:
-        # Note: If both token and email were passed in, it would be possible
-        # that subscribers exist that match one or the other but not both.
-        # But currently no callers pass both, so luckily we don't have to
-        # figure out what we would do in that case.
+
+    # Note: If both token and email were passed in, it would be possible
+    # that subscribers exist that match one or the other but not both.
+    # But currently no callers pass both, so luckily we don't have to
+    # figure out what we would do in that case.
```

Comment:
This comment indicates an intentionally unhandled edge case, which is slightly worrying, especially since `update_user_task` in `news/views.py` passes in both the email and the token that it got from user-submitted data.